receiver/envoyalsreceiver/internal/als/server.go

```diff
@@ -0,0 +1,89 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+package als // import "github.com/open-telemetry/opentelemetry-collector-contrib/receiver/envoyalsreceiver/internal/als"
+
+import (
+	"context"
+	"errors"
+	"io"
+
+	alsv3 "github.com/envoyproxy/go-control-plane/envoy/service/accesslog/v3"
+	"go.opentelemetry.io/collector/consumer"
+	"go.opentelemetry.io/collector/pdata/pcommon"
+	"go.opentelemetry.io/collector/pdata/plog"
+	"go.opentelemetry.io/collector/receiver/receiverhelper"
+)
+
+type Server struct {
+	nextConsumer consumer.Logs
+	obsrep       *receiverhelper.ObsReport
+}
+
+func New(nextConsumer consumer.Logs, obsrep *receiverhelper.ObsReport) *Server {
+	return &Server{
+		nextConsumer: nextConsumer,
+		obsrep:       obsrep,
+	}
+}
+
+func (s *Server) StreamAccessLogs(logStream alsv3.AccessLogService_StreamAccessLogsServer) error {
+	for {
+		data, err := logStream.Recv()
+		if errors.Is(err, io.EOF) {
+			break
+		}
+		if err != nil {
+			return err
+		}
+
+		ctx := s.obsrep.StartLogsOp(context.TODO())
+		logs := toLogs(data)
+		logRecordCount := logs.LogRecordCount()
+		err = s.nextConsumer.ConsumeLogs(ctx, logs)
+		s.obsrep.EndLogsOp(ctx, "protobuf", logRecordCount, err)
+		if err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+const (
+	apiVersionAttr = "api_version"
+	apiVersionVal  = "v3"
+	logTypeAttr    = "log_type"
+	httpTypeVal    = "http"
+	tcpTypeVal     = "tcp"
+)
```

Comment:
I understand the argument for having these local to the function where they're used, but could we move them to the top of the file? I think constants are conventionally always declared in a block at the start of the file.

The alternative here could be to declare them in `toLogs`.